pkg/acpi/acpi.go

```diff
@@ -0,0 +1,219 @@
+// had gotten this right, but that code is pretty bad too).
+// What's really fun: when you generate the [RX]SDT, you need to generate pointers
+// too. It's best to keep those tables in a physically contiguous area, so we do;
+// not all systems follow this rule, which is asking for trouble.
+// Finally, not all tables are generated in a consistent way, e.g. the IBFT is not like
+// most other tables containing variable length data, because the people who created it
+// are Klever. IBFT has its own heap, and the elements in the heap follow rules unlike
+// all other tables. Nice!
+// If you really look at ACPI closely, you can kind of see that it's optimized for NASM,
+// which is part of what makes it so unpleasant. But, it's what we have.
+package acpi
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"log"
+	"strconv"
+)
+
+const (
+	// LengthOffset is the offset of the table length
+	LengthOffset = 4
+	// CSUMOffset is the offset of the single byte checksum in *most* ACPI tables
+	CSUMOffset = 9
+	// MinTableLength is the minimum length: 4 byte tag, 4 byte length, 1 byte revision, 1 byte checksum,
+	MinTableLength = 10
+)
+
+var (
+	// Debug implements fmt.Sprintf and can be used for debug printing
+	Debug        = func(string, ...interface{}) {}
+	unmarshalers = map[sig]func(Tabler) (Tabler, error){}
+)
+
+// addUnMarshaler is intended to be called by init functions
+// in this package. It adds an UnMarshaler for a given
+// ACPI signature.
+func addUnMarshaler(n string, f func(Tabler) (Tabler, error)) {
+	if _, ok := unmarshalers[sig(n)]; ok {
+		log.Fatalf("Can't add %s; already in use", n)
+	}
+	unmarshalers[sig(n)] = f
+}
+
+// GetHeader extracts a Header from a Tabler and returns a reference to it.
+func GetHeader(t Tabler) *Header {
+	return &Header{
+		Sig:             sig(t.Sig()),
+		Length:          t.Len(),
+		Revision:        t.Revision(),
+		CheckSum:        t.CheckSum(),
+		OEMID:           oem(t.OEMID()),
+		OEMTableID:      tableid(t.OEMTableID()),
+		OEMRevision:     t.OEMRevision(),
+		CreatorID:       t.CreatorID(),
+		CreatorRevision: t.CreatorRevision(),
+	}
+}
+
+// Flags takes 0 or more flags and produces a uint32 value.
+// Each argument represents one bit position, with the first flag
+// being bit 0. For each flag with a value of "1", the bit for that
+// flag will be set. For each flag with a value of "0", that bit in
+// the flag will be cleared.
+// This is mainly for consistency but it would allow
+// us in future to pass in an initial value and set or clear bits in it
+// depending on flags.
+// Current allowed values are "0" and "1". In future, if the flags are
+// not contiguous, we can allow "ignore" in future to ignore a flag.
+func flags(s ...flag) (uint8, error) {
+	var i, bit uint8
+	for _, f := range s {
+		switch f {
+		case "1":
+			i |= 1 << bit
+		case "0":
+			i &= ^(1 << bit)
+		default:
+			return 0, fmt.Errorf("%s is not a valid value: only 0 or 1 are valied", f)
```

Comment:
```suggestion
			return 0, fmt.Errorf("%s is not a valid value: only 0 or 1 are valid", f)
```

Reply:
done. 

I love how github scrollbar hides most of your suggestion :-)